src/core/error_helpers.js

```diff
@@ -105,6 +110,159 @@ p5._friendlyFileLoadError = function (errorType, filePath) {
   report(message, errorInfo.method, FILE_LOAD);
 };
 
+/**
+* Validates parameters
+* param  {String}               func    the name of the function
+* param  {Array}                args    user input arguments
+*
+* example:
+*  var a;
+*  ellipse(10,10,a,5);
+* console ouput:
+*  "It looks like ellipse received an empty variable in spot #2."
+*
+* example:
+*  ellipse(10,"foo",5,5);
+* console output:
+*  "ellipse was expecting a number for parameter #1,
+*           received "foo" instead."
+*/
+function validateParameters(func, args) {
+  var arrDoc = lookupParamDoc(func);
+  if (arrDoc.length > 1){   // multiple format?
+    var errorListArray = [];
+    for (var i = 0; i < arrDoc.length; i++) {
+      errorListArray.push(testParamFormat(args, arrDoc[i]));
+    }
+    // compare errors from all formats
+    var minErrInd = -1;
+    var minErrCount = 0;
+    for (var j = 0; j < errorListArray.length; j++) {
+      var numErr = errorListArray[j].length;
+      if (numErr >= arrDoc.length){   // exclude clean cases
+        if(j===0){  // the first case
+          minErrCount = numErr;
+          minErrInd = j;
+        }else{
+          if(minErrCount > numErr){
+            minErrInd = j;
+          }
+        }
+      }
+    }
+    if (minErrInd > -1){
+      // generate error msg with a smaller number of errors
+      for(var n = 0; n < errorListArray[minErrInd].length; n++) {
+        p5._friendlyParamError(errorListArray[minErrInd][n], func);
+      }
+    }
+  } else {
+    var errorArray = testParamFormat(args, arrDoc[0]);
+    if (errorArray.length > 0){
+      for(var m = 0; m < errorArray.length; m++) {
+        p5._friendlyParamError(errorArray[m], func);
+      }
+    }
+  }
+}
+// validateParameters() helper functions:
+// lookupParamDoc() for querying data.json
+function lookupParamDoc(func){
+  var queryResult = arrDoc.classitems.
+    filter(function (x) { return x.name === func; });
+  if (queryResult[0].hasOwnProperty('overloads')){
+    var res = [];
+    for(var i = 0; i < queryResult[0].overloads.length; i++) {
+      res.push(queryResult[0].overloads[i].params);
+    }
+    return res;
+  } else {
+    return [queryResult[0].params];
+  }
+}
+function testParamFormat(args, format){
+  var errorArray = [];
+  var error;
+  for (var p = 0; p < format.length; p++) {
+    var argType = typeof(args[p]);
+    if ('undefined' === argType || null === argType) {
+      if (format[p].optional !== true) {
+        error = {type:'EMPTY_VAR', position: p};
+        errorArray.push(error);
+      }
+    } else {
+      var types = format[p].type.split('|'); // case accepting multi-types
+      var pass;
+      if (argType === 'object'){             // if obejct, test for class
+        pass = testParamClass(args[p], types);
+        if (!pass) {  // if fails to pass
```

Comment:
we could remove the creation of the variable pass, and just rewrite this line as
`if (!testParamClass(args[p], types)) {`
same with the other one in the else statement below